### PR TITLE
Flesh out skeleton for AMT calls

### PR DESCRIPTION
**Pull request by Berrescuda**
This update just splits the AMT calls we can make into two devices and adds the space where the actual logic should go for the power on/off/cycle/reset commands and boot target setting.

### CODE REVIEW

**Review by KylerBurke — APPROVED** [review]

File: devices/boot_target.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/vapor-ware/synse-sdk/sdk"
+	"github.com/vapor-ware/synse-sdk/sdk/logger"
+)
+
+// AmtBootTarget handler for setting an amt device's boot target.
```
Comment (edaniszewski) [inline devices/boot_target.go:11]: AmtBootTarget is the handler...

File: devices/boot_target.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+func bootTargetWrite(device *sdk.Device, data *sdk.WriteData) error {
+	action := data.Action
+	raw := data.Raw
+	// When writing to a boot_target device, we always expect there to be
+	// raw data specified. If there isn't, we return an error.
+	if len(raw) == 0 {
+		return fmt.Errorf("no values specified for 'raw', but required")
+	}
+
+	if action == "state" {
```
Comment (edaniszewski) [inline devices/boot_target.go:38]: boot target should take "target" as action, not state

File: devices/boot_target.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+			logger.Debug("Set Boot Target to cd")
+		case "hd":
+			logger.Debug("Set Boot Target to hd")
+		default:
+			return fmt.Errorf("unsupported command for amt power 'state' action: %s", cmd)
+		}
+
+	} else {
+		// If we reach here, then the specified action is not supported.
+		return fmt.Errorf("action '%s' is not supported for bmc power devices", action)
```
Comment (edaniszewski) [inline devices/boot_target.go:52]: no BMC for AMT

should probably also say 'boot target devices'

File: devices/boot_target.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+// bootTargetRead gets the current power state of the AMT device
+func bootTargetRead(device *sdk.Device) ([]*sdk.Reading, error) {
+
+	readings := []*sdk.Reading{
+		sdk.NewReading("state", "default"),
+	}
+	return readings, nil
+}
+
+func bootTargetWrite(device *sdk.Device, data *sdk.WriteData) error {
```
Comment (edaniszewski) [inline devices/boot_target.go:29]: no function comment

File: devices/boot_target.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+var AmtBootTarget = sdk.DeviceHandler{
+	Type:  "boot_target",
+	Model: "amt-boot-target",
+
+	Read:  bootTargetRead,
+	Write: bootTargetWrite,
+}
+
+// bootTargetRead gets the current power state of the AMT device
+func bootTargetRead(device *sdk.Device) ([]*sdk.Reading, error) {
```
Comment (edaniszewski) [inline devices/boot_target.go:21]: not power state, boot target state

File: devices/power.go [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+// amtPowerRead gets the current power state of the AMT device
+func amtPowerRead(device *sdk.Device) ([]*sdk.Reading, error) {
+
+	readings := []*sdk.Reading{
+		sdk.NewReading("state", "on"),
+	}
+	return readings, nil
+}
+
+func amtPowerWrite(device *sdk.Device, data *sdk.WriteData) error {
```
Comment (edaniszewski) [inline devices/power.go:29]: comment

File: devices/power.go [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+			logger.Debug("AMT Power Resetting")
+		case "cycle":
+			logger.Debug("AMT Power Cycling")
+		default:
+			return fmt.Errorf("unsupported command for amt power 'state' action: %s", cmd)
+		}
+
+	} else {
+		// If we reach here, then the specified action is not supported.
+		return fmt.Errorf("action '%s' is not supported for bmc power devices", action)
```
Comment (edaniszewski) [inline devices/power.go:56]: no bmcs for amt

**Pull request merged**